rocAL/rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -77,12 +77,13 @@ ROCAL_API_CALL rocalCreateVideoLabelReader(RocalContext p_context, const char* s
 }
 
 RocalMetaData
-ROCAL_API_CALL rocalCreateCOCOReader(RocalContext p_context, const char* source_path, bool is_output){
+ROCAL_API_CALL rocalCreateCOCOReader(RocalContext p_context, const char* source_path, bool is_output, const bool mask) {
     if (!p_context)
         THROW("Invalid rocal context passed to rocalCreateCOCOReader")
     auto context = static_cast<Context*>(p_context);
-
-    return context->master_graph->create_coco_meta_data_reader(source_path, is_output, MetaDataReaderType::COCO_META_DATA_READER,  MetaDataType::BoundingBox);
+    if(mask)
+        return context->master_graph->create_coco_meta_data_reader(source_path, is_output, MetaDataReaderType::COCO_META_DATA_READER, MetaDataType::PolygonMask);
```

Comment:
can't we read the polygons mask and bounding boxes from the same api. Going through the dataset twice is inefficient

Reply:
The coco_meta_data_reader reads both polygon masks and bboxes in a single read. Depending on MetaDataType passed to  create_coco_meta_data_reader, it either reads boxes only or boxes and polygon masks